library/std/src/sys/unix/fs.rs

```diff
@@ -962,6 +963,12 @@ pub fn rename(old: &Path, new: &Path) -> io::Result<()> {
     Ok(())
 }
 
+pub fn get_openoptions_as_cint(from: OpenOptions) -> io::Result<libc::c_int> {
```

Comment:
This should be an as_flags method on the existing extension trait OpenOptionsExt.

It should operate on &self.